pkg/operator/hive/hive_controller.go

```diff
@@ -513,6 +514,15 @@ func (r *ReconcileHiveConfig) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	// Read the metricsConfig section and fetch clusterDeploymentLabelSelectors, error out if it fails validation
+	err = controllerMetrics.GetClusterDeploymentLabelSelectors(hLog, origHiveConfig.Spec.MetricsConfig)
+	if err != nil {
+		hLog.WithError(err).Error("error in metricsConfig")
+		instance.Status.Conditions = util.SetHiveConfigCondition(instance.Status.Conditions, hivev1.HiveReadyCondition, corev1.ConditionFalse, "ErrorInMetricsConfig", err.Error())
+		r.updateHiveConfigStatus(origHiveConfig, instance, hLog, false)
+		return reconcile.Result{}, err
+	}
+
```

Comment:
Is this location/incorporation correct? My testing hasn't yielded fruitful results - happy to move this to metrics controller and panic if error, and tackle this feature later if time is an issue